src/httpserver/webserver.hpp

```diff
@@ -95,19 +95,20 @@ class webserver
         **/
         bool is_running();
         /**
-         * Method used to registrate a resource to the webserver.
+         * Method used to register a resource with the webserver.
          * @param resource The url pointing to the resource. This url could be also parametrized in the form /path/to/url/{par1}/and/{par2}
          *                 or a regular expression.
          * @param http_resource http_resource pointer to register.
          * @param family boolean indicating whether the resource is registered for the endpoint and its child or not.
+         * @return true if the resource was registered
         **/
         template <typename T>
-        void register_resource(const std::string& resource,
+        bool register_resource(const std::string& resource,
                 http_resource<T>* res, bool family = false
         )
         {
             details::http_resource_mirror hrm(res);
-            register_resource(resource, hrm, family);
+            return register_resource(resource, hrm, family);
```

Comment:
I was trying to track down an issue with resource registration when I noticed the status wasn't being returned here.